src/integrationTest/sam.test.ts

```diff
@@ -397,246 +397,206 @@ describe('SAM Integration Tests', async function () {
+                cfnTemplatePath = path.join(runtimeTestRoot, samApplicationName, 'template.yaml')
+                readmePath = path.join(runtimeTestRoot, samApplicationName, 'README.md')
+
                 console.log('runtimeTestRoot: ', runtimeTestRoot)
                 mkdirpSync(runtimeTestRoot)
+                await createSamApplication(runtimeTestRoot, scenario)
+                await verifySamAppExists(cfnTemplatePath, readmePath) // sanity check
+                samAppCodeUri = await openSamAppFile(appPath)
             })
 
-            after(async function () {
-                // don't clean up after java tests so the java language server doesn't freak out
-                if (scenario.language !== 'java') {
-                    await tryRemoveFolder(runtimeTestRoot)
-                }
+            beforeEach(async function () {
+                testDisposables = []
+                await closeAllEditors()
             })
 
-            function log(o: any) {
-                console.log(`sam.test.ts: scenario ${scenarioIndex} (${scenario.displayName}): ${o}`)
-            }
-
-            /**
-             * This suite cleans up at the end of each test.
-             */
-            describe('Starting from scratch', async function () {
-                let testDir: string
-
-                beforeEach(async function () {
-                    testDir = await mkdtemp(path.join(runtimeTestRoot, 'test-'))
-                    log(`testDir: ${testDir}`)
-                })
-
-                afterEach(async function () {
-                    // don't clean up after java tests so the java language server doesn't freak out
-                    if (scenario.language !== 'java') {
-                        await tryRemoveFolder(testDir)
-                    }
-                })
-
-                it('creates a new SAM Application (happy path)', async function () {
-                    await createSamApplication(testDir)
-
-                    // Check for readme file
-                    const readmePath = path.join(testDir, samApplicationName, 'README.md')
-                    assert.ok(await fileExists(readmePath), `Expected SAM App readme to exist at ${readmePath}`)
-                })
+            afterEach(async function () {
+                testDisposables.forEach(d => d.dispose())
+                await stopDebugger(undefined)
             })
 
-            /**
-             * This suite makes a sam app that all tests operate on.
-             * Cleanup happens at the end of the suite.
-             */
-            describe(`Starting with a newly created ${scenario.displayName} SAM Application...`, async function () {
-                let testDisposables: vscode.Disposable[]
-
-                let testDir: string
-                let samAppCodeUri: vscode.Uri
-                let appPath: string
-                let cfnTemplatePath: string
+            after(async function () {
+                await tryRemoveFolder(runtimeTestRoot)
```

Comment:
Are we able to safely remove the java check:
```typescript
if (scenario.language !== 'java') {
    await tryRemoveFolder(testDir)
}
```
I'm guessing the problem only appeared because we were creating multiple sam apps and now we're only creating one?

Reply:
I think this logic was a hack to make the Java codelens tests work. Preserving the directory gave the Java extension time to provide document symbols. Try adding a long sleep before running the failing tests to verify this. If it works then we can figure out a better solution.